draft-ietf-quic-manageability.md

```diff
@@ -1023,6 +1025,53 @@ not arrive at their destination. Admitting a few packets allows the QUIC
 endpoint to determine that the path accepts QUIC. Sudden drops afterwards will
 result in slow and costly timeouts before abandoning the connection.
 
+## Guiding Path MTU
+
+Some networks support 1500-byte packets, but can only do so by fragmenting at a
+lower layer before traversing a smaller MTU segment, and then reassembling.
+This is permissible even when the IP layer is IPv6 or IPv4 with the DF bit set,
+because it occurs below the IP layer. However, this process can add to compute
+and memory costs, leading to a bottleneck that limits network capacity. In such
+networks this generates a desire to influence a majority of senders to use
+smaller packets, so that the limited reassembly capacity is not exceeded.
+
+For TCP, MSS clamping ({{Section 3.2 of RFC4459}}) is often used to change
+the sender's maximum TCP segment size, but QUIC requires a different approach.
+{{Section 14 of QUIC-TRANSPORT}} advises senders to probe larger sizes using
+Datagram Packetization Layer PMTU Discovery ({{?DPLPMTUD=RFC8899}}) or Path
+Maximum Transmission Unit Discovery (PMTUD: {{?RFC1191}} and {{?RFC8201}}).
+This mechanism will encourage senders to approach the maximum size, which
+could cause fragmentation with a network segment that they may not be aware of.
+
+If path performance is limited when sending larger packets, an on-path
+device should support a maximum packet size for a specific transport flow
+and then consistently drop all packets that exceed the configured size
+when the inner IPv4 packet has DF set, or IPv6 is used. Endpoints can cache
+PMTU information between IP flows, in the IP-layer cache, so short-term
+consistency between the PMTU for flows can help avoid an endpoint using a
+PMTU that is inefficient.
+
+If network operators wish to start to drop large packets to avoid
+fragmentation, they may wish to focus on QUIC. QUIC packets cannot always
+be identified ({{sec-identifying}}), but we assume that a set of candidate
+QUIC flows have been chosen. For QUIC endpoints using DPLPMTUD it is
+minimally sufficient for the path to silently drop a packet larger than the
+supported size. (A QUIC probe packet is used to discover the PMTU. If lost,
+this does not impact the flow of QUIC data.)
```

Comment:
```suggestion
Dropping QUIC packets that are used by DPLPMTUD to probe the path MTU will
not impact QUIC performance but can prevent a QUIC flow from increasing its packet
size.
Even though QUIC packets cannot be unambiguously identified ({{sec-identifying}} as
probes, network with configurations that would lead to fragmentation of large packets 
should rather drop large QUIC packet than fragmenting them as this will likely provide
the right signal to the endpoint to keep the packet size small and thereby avoid network
fragmentation for that flow entirely.
```

Reply:
Mirja I can see the value in amending the wording from "does not impact the flow of QUIC data" to "will not impact QUIC performance but can prevent a QUIC flow from increasing its packet size". Happy with that. But what your suggested change removes are a couple of notions. One is that it's a valid/common operator choice to be cautious in deploying packet drops, by choosing to focus on something well-known (QUIC, in this case) rather than all large IP. Secondly, the notion that QUIC itself can be hard to identify. It's not always as obvious as UDP/443. You refer instead to the ability to identify probe packets from within a known QUIC flow. Are you explicitly wanting to remove both of these, or are these merely side-effects of your suggested change?

Reply:
I wasn't trying to change the content here but to make it more clear what the recommendation is. I was working on the slides for the meeting and was looking for one sentence to copy in the slides that actually makes a recommendation (like my second proposed sentence now) but could find it.

I didn't realise that you are also talking about identifying QUIC (with respect to other UDP traffic) as you talk about QUIC packets and not QUIC flows in the original text but I left the reference to sec-identifying which should give the reader enough context anyway.

The recommendation explicitly says "QUIC packets" so I thought it was sufficiently clear that we basically say "yes we understand that you want to careful about dropping packets but if for QUIC is would be really good to drop anyway". I guess we could add back another sentence to make that bit more clear. Again my intention really was to make more clear what the recommendation actually is; so was looking for a sentence with "should" in it ;-)

Reply:
You'll find two sentences with recommendations ("should") at the end.

As for this particular paragraph, I propose the following merged form.

> Networks with configurations that would lead to fragmentation of large packets should drop such packets rather than fragmenting them. If network operators wish to start implementing this, they may wish to focus on QUIC. QUIC flows cannot always be identified ({{sec-identifying}}), but we assume that a set of candidate flows have been chosen. For QUIC endpoints using DPLPMTUD it is minimally sufficient for the path to silently drop a packet larger than the supported size. (A QUIC probe packet is used to discover the PMTU. If lost, this does not impact the flow of QUIC data.)

I'm trying to avoid saying networks should _just_ drop these packets. The recommendation is to drop and send PTB. But sometimes PTB won't be possible.

So I also propose using your sentence in the final paragraph:

> Since a network cannot know in advance which discovery method a QUIC endpoint is using, it should always send a PTB message in addition to dropping the oversized packet. A generated PTB message should be compliant with the validation requirements of {{Section 14.2.1 of QUIC-TRANSPORT}}, otherwise it will be ignored by DPLPMTUD. This will likely provide the right signal for the endpoint to keep the packet size small and thereby avoid network fragmentation for
> that flow entirely.

Reply:
Happy to do nearly all the above. The only part I don't think is needed are "based on e.g. handshake semantics or the connection ID". I'll add a commit after the meeting.

Reply:
Okay. Thx!